provider/core/src/resource.rs

```diff
@@ -142,57 +142,46 @@ impl ResourceKey {
         // Regex: [a-zA-Z0-9=_]+(/[a-zA-Z0-9=_]+)*@[0-9]+
         enum State {
             Start,
-            AfterChar,
-            AfterSlash,
-            AfterCharAfterSlash,
-            AfterAt,
-            AfterDigit,
+            Body0,
```

Comment:
suggestion (nb): while you're here, do you want to add some doc comments explaining each state qualitatively?